File: modules/wordOfTheDay.py
import requests
from bs4 import BeautifulSoup
import logging


def __word_of_the_day():
    union_berlin_tickets_url = "https://www.nytimes.com/column/learning-word-of-the-day"
    headers = {'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:82.0) Gecko/20100101 Firefox/82.0'}
    html_text = requests.get(union_berlin_tickets_url, timeout=5, headers=headers).text
    soup = BeautifulSoup(html_text, 'html.parser')

    last_article = soup.find('article')
    last_article_url = "https://www.nytimes.com" + last_article.find('a')['href']
    last_article_title = last_article.find('h3').text

    res =f"[{last_article_title}]({last_article_url}):\n"

    return res


def word_of_the_day() -> str:
    try:
        return __word_of_the_day()
    except:
        logging.exception("Couldn't get word of the day")
        return "Couldn't get word of the day"

File: app/modules/wordOfTheDay.go
package modules

import (
	"fmt"
	"log"
	utils "main/utils"
)

func WordOfTheDay() *WordOfTheDayData {
	res, err := wordOfTheDay()
    if err != nil {
		log.Printf("Error in Word of the day module: %s", err)
	 }
	return res
}

func wordOfTheDay() (*WordOfTheDayData, error) {
	wordOfTheDayURL := "https://www.nytimes.com/column/learning-word-of-the-day"
	doc, err := utils.GetDoc(wordOfTheDayURL)
	if err != nil {
		return nil, err
	}

	lastArticle := doc.Find("article")
	lastArticleURL, _ := lastArticle.Find("a").First().Attr("href")
	lastArticleTitle := lastArticle.Find("h3").First().Text()

	res := fmt.Sprintf("[%s](https://www.nytimes.com%s)\n", lastArticleTitle, lastArticleURL)

	return &WordOfTheDayData{res}, nil
}

type WordOfTheDayData struct {
	data string
}

func (d *WordOfTheDayData) String() string {
	if d == nil {
		return ""
	}
    return d.data
}